rasa/train.py

```diff
@@ -222,17 +229,37 @@ async def _do_training(
     persist_nlu_training_data: bool = False,
     core_additional_arguments: Optional[Dict] = None,
     nlu_additional_arguments: Optional[Dict] = None,
+    old_model_zip_path: Optional[Text] = None,
 ):
     if not fingerprint_comparison_result:
         fingerprint_comparison_result = FingerprintComparisonResult()
 
+    interpreter_path = None
+    if fingerprint_comparison_result.should_retrain_nlu():
+        model_path = await _train_nlu_with_validated_data(
+            file_importer,
+            output=output_path,
+            train_path=train_path,
+            fixed_model_name=fixed_model_name,
+            persist_nlu_training_data=persist_nlu_training_data,
+            additional_arguments=nlu_additional_arguments,
+        )
+        interpreter_path = os.path.join(model_path, DEFAULT_NLU_SUBDIRECTORY_NAME)
```

Comment:
I think `model_path` can be a `*.tar.gz` path. In this case `os.path.join()` will not provide a proper path, I think.

Reply:
(And I have a feeling unpacking it here, just after saving, is a bit of a stupid solution)

Reply:
it can return a `tar.gz`, but only when NLU was trained on its own: https://github.com/RasaHQ/rasa/blob/bd8cc1e633e604ca505ab0c857fe2bd2b932d8fc/rasa/train.py#L543